src/main/java/core/basesyntax/strategy/OperationStrategy.java

```diff
@@ -0,0 +1,33 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.impl.BalanceOperation;
+import core.basesyntax.strategy.impl.PurchaseOperation;
+import core.basesyntax.strategy.impl.ReturnOperation;
+import core.basesyntax.strategy.impl.SupplyOperation;
+
+public class OperationStrategy {
+    private static final String BALANCE_NAME = "b";
+    private static final String PURCHASE_NAME = "p";
+    private static final String RETURN_NAME = "r";
+    private static final String SUPPLY_NAME = "s";
+
+    public void implementOperationStrategy(FruitTransaction fruitTransaction) {
+        switch (fruitTransaction.getOperationType().getName()) {
```

Comment:
[Avoid using switch-cases and if-else constructions. It is recommended to use the Strategy pattern instead](https://mate-academy.github.io/jv-program-common-mistakes/java-core/solid/fruit-shop#avoid-using-switch-cases-and-if-else-constructions-it-is-recommended-to-use-the-strategy-pattern-instead)